packages/api/src/post-build.js

```diff
@@ -105,20 +135,50 @@ class PostBuildHandler {
     }
   }
 
+  static async flattenTap (datastring) {
+    const output = await new Promise(resolve => {
+      var process = cp.exec('tap-parser -f -t', (error, stdout, stderr) => {
+        if (error || stderr) {
+          throw new InvalidParameterError('Could not parse and flatten tap');
+        }
+        resolve(stdout);
+      });
+
+      // write your multiline variable to the child process
+      process.stdin.write(datastring);
+      process.stdin.end();
+    });
+    return output.replace(/\d+..\d+\n/, '\n'); // hacky solution to issue where 1..0 shows up at beginning
+  }
+
+  static removeDuplicateTestCases (testCases) {
```

Comment:
Does this take into account class names? For example, two different directories may have tests with the same name.

Reply:
Most TAP generators would prefix the test names to be `class/test name`, but other than that there is no way to know since this endpoint accepts just one TAP file from all classes aggregated without class information